tests/nnstreamer_filter_mvncsdk2/NCSDKTensorFilterTestHelper.cc

```diff
@@ -0,0 +1,564 @@
+      unsigned int *dataLength)
+{
+  switch (option) {
+    case NC_RO_API_VERSION:
+      if (sizeof(ncsdk_ver_t) != (*dataLength))
+        return NC_ERROR;
+      memcpy (data, this->mVer, *dataLength);
+      break;
+    default:
+      return NC_ERROR;
+  }
+  return NC_OK;
+}
+
+/**
+ * @brief A method mocking ncDeviceCreate()
+ */
+ncStatus_t
+NCSDKTensorFilterTestHelper::ncDeviceCreate (int index,
+    struct ncDeviceHandle_t **deviceHandle)
+{
+  if ((index < 0) || (index >= SUPPORT_MAX_NUMS_DEVICES))
+    return NC_INVALID_PARAMETERS;
+
+  if (this->mDevHandle == nullptr)
+    return NC_OUT_OF_MEMORY;
+
+  *deviceHandle = this->mDevHandle;
+
+  return NC_OK;
+}
+
+/**
+ * @brief A method mocking ncDeviceOpen()
+ */
+ncStatus_t
+NCSDKTensorFilterTestHelper::ncDeviceOpen(struct ncDeviceHandle_t *deviceHandle)
+{
+  if (deviceHandle != this->mDevHandle)
+    return NC_NOT_ALLOCATED;
+
+  return NC_OK;
+}
+
+/**
+ * @brief A method mocking ncDeviceClose()
+ */
+ncStatus_t
+NCSDKTensorFilterTestHelper::ncDeviceClose(
+    struct ncDeviceHandle_t *deviceHandle)
+{
+  if (deviceHandle != this->mDevHandle)
+    return NC_NOT_ALLOCATED;
+
+  return NC_OK;
+}
+
+/**
+ * @brief A method mocking ncDeviceDestroy()
+ */
+ncStatus_t
+NCSDKTensorFilterTestHelper::ncDeviceDestroy(
+    struct ncDeviceHandle_t **deviceHandle)
+{
+  if ((this->mDevHandle != nullptr) && (*deviceHandle != nullptr)
+      && (*deviceHandle == this->mDevHandle)) {
+    delete this->mDevHandle;
+    this->mDevHandle = nullptr;
+    *deviceHandle = nullptr;
+  } else {
+    return NC_NOT_ALLOCATED;
+  }
+
+  return NC_OK;
+}
+
+/**
+ * @brief A method mocking ncGraphCreate()
+ */
+ncStatus_t
+NCSDKTensorFilterTestHelper::ncGraphCreate(const char* name,
+    struct ncGraphHandle_t **graphHandle)
+{
+  if (!g_file_test (name, G_FILE_TEST_IS_REGULAR)) {
+    return NC_INVALID_PARAMETERS;
+  }
+
+  if (this->mGraphHandle == nullptr) {
+    return NC_OUT_OF_MEMORY;
+  }
+
+  this->mModelPath = g_strdup (name);
+  *graphHandle = this->mGraphHandle;
+
+  return NC_OK;
+}
+
+/**
+ * @brief A method mocking ncGraphAllocate()
+ */
+ncStatus_t
+NCSDKTensorFilterTestHelper::ncGraphAllocate(
+    struct ncDeviceHandle_t *deviceHandle, struct ncGraphHandle_t *graphHandle,
+    const void *graphBuffer, unsigned int graphBufferLength)
+{
+  if ((this->mDevHandle != deviceHandle) || (this->mGraphHandle != graphHandle))
+    return NC_INVALID_PARAMETERS;
+
+  this->mGraphBuf = graphBuffer;
+  this->mLenGraphBuf = graphBufferLength;
+
+  return NC_OK;
+}
+
+/**
+ * @brief A method mocking ncGraphGetOption().
+ * A description of input or output tensor will be filled in data.
+ */
+ncStatus_t
+NCSDKTensorFilterTestHelper::ncGraphGetOption(
+    struct ncGraphHandle_t *graphHandle, int option, void *data,
+    unsigned int *dataLength)
+{
+  switch (option) {
+    case NC_RO_GRAPH_INPUT_TENSOR_DESCRIPTORS:
+      if (this->mTensorDescInput == nullptr) {
+        return NC_OUT_OF_MEMORY;
+      }
```

Comment:
Does this need to check `*dataLength` with `ncTensorDescriptor_t` (like line 123)?

Reply:
It does. I added the checking code.